### DISCUSSION TITLE
Installation options for macOS and Windows?

### DISCUSSION THREAD

**Discussion by ylluminate** [input-leap/input-leap#1656] [Q&A]
I'd like to shift over from Barrier to Input Leap for some workstations, but I'm not seeing a clear path for installation at this point for macOS, Windows and Linux on the front page (README).

If there are some clear paths to follow, could someone please elucidate further?

So far I'm seeing the following info:
* https://github.com/input-leap/input-leap/wiki
* https://github.com/vancluever/homebrew-input-leap

It seems from some advice from @shymega that we can also use artifacts from successful builds from the CI:
* https://github.com/input-leap/input-leap/actions?query=is%3Asuccess
  * such as this: https://github.com/input-leap/input-leap/actions/runs/4714622749

**Comment by jrmann1999** [comment]
I can't find any builds with MacOS artifacts, it appears they build, but are not retained.

**Reply by cvladan** [reply]
Exactly

**Reply by shymega** [reply]
CI PR is waiting for approval. This will retain macOS DMGs. But not M1 yet.

**Comment by shymega** [comment]
I'm waiting for a PR that revamps CI to be merged. You can ignore Azure Pipelines as well, I want to move to GH Actions completely.

I think Azure does generate macOS DMGs...

I also wouldn't install any builds right now. There's still a few critical bugs, which need to be fixed.

**Comment by dlozynski** [comment]
@jrmann1999 
macOS build you can find in this discussion [Unofficial Homebrew tap for MacOS #1623](https://github.com/input-leap/input-leap/discussions/1623) 
for windows it can be found in Actions of this project, at the very bottom , I remember that the links in one browser I had it blocked because of some ads-extention. try this link [https://github.com/input-leap/input-leap/suites/12269501649/artifacts/649703494](https://github.com/input-leap/input-leap/suites/12269501649/artifacts/649703494)
all builds are in debug and some debug libs are required and on windows I did not had a chance to test it yet

@ylluminate thanks for pointing out about links, corrected

**Comment by ylluminate** [comment]
@dlozynski fwiw your links both go to wrong urls, here are the outright URLs:
- https://github.com/input-leap/input-leap/discussions/1623
- https://github.com/input-leap/input-leap/suites/12269501649/artifacts/649703494

**Comment by Rings4** [comment]
> @dlozynski fwiw your links both go to wrong urls, here are the outright URLs:
> 
>     * [Unofficial Homebrew tap for MacOS #1623](https://github.com/input-leap/input-leap/discussions/1623)
> 
>     * https://github.com/input-leap/input-leap/suites/12269501649/artifacts/649703494

How do you actually run this? It's saying I'm missing some pre-requisites and to re-install it, but I don't know how to do that since there is no installer. I'm on Windows.

**Reply by ylluminate** [reply]
I think you might be missing the Windows artifact link above from @dlozynski. Whether that will solve your problems, I don't know, but there's a chance you'll get some relief and worth trying to download it to see.

**Reply by Rings4** [reply]
> I think you might be missing the Windows artifact link above from @dlozynski. Whether that will solve your problems, I don't know, but there's a chance you'll get some relief and worth trying to download it to see.

No, I'm not missing that link. This one, right? https://github.com/input-leap/input-leap/suites/12269501649/artifacts/649703494 Like I said, it's saying I'm missing some pre-requisites and to re-install it, but I don't know how to do that since there is no installer.

**Reply by shymega** [reply]
Waiting for feedback on a CI PR review, then the installer will be built at the same time. Right now, there's no way to do this.

**Reply by cforce** [reply]
Any news?

**Reply by shymega** [reply]
I did prod @p12tic via email recently but haven't heard back yet. I can't merge until the review is approved.

**Reply by ylluminate** [reply]
Anything we can help you with @p12tic to move things ahead?

**Reply by shymega** [reply]
Nope. I think they're busy with contracts. I will chase again...

**Comment by jrmann1999** [comment]
Checking in, any updates?

**Reply by shymega** [reply]
Hi,

Please avoid bumping the issue when unnecessary.

If you also experience the issue, kindly react to the issue with `:+1:`, or if you want to find out when the issue will be resolved, please do not bump the issue, but _instead_ subscribe to the issue using the button on the right-hand side.

**Reply by shymega** [reply]
To clarify, sorry, that's a saved reply I use.

I have exchanged a few emails with Povilas, but it was mostly regarding the Rust side of things (which the FFI branch has been moved to my fork, and the rewrite to my account), and other unrelated things.

I can't say when v3.0.0 will be released, but I'd like to get the CI sorted.

**Comment by ann0see** [comment]
I believe, you can just fork the repo and run the build on GH actions yourself and check the release checkbox in the actions tab. I think it would be good to provide compiled beta binaries somewhere as this enables more and lower hurdle testing.

**Reply by shymega** [reply]
Yes, this is one way around it. We can't do beta binaries right now.

**Comment by sithlord48** [comment]
https://github.com/input-leap/input-leap/discussions/1793#discussioncomment-8577620
